### PR TITLE
problem: token sending not working

### PR DESCRIPTION
**Pull request by shanejonas**
solution: refactor create transaction ui to `emerald-js-ui` + upgrade to react 16

### CODE REVIEW

File: src/components/tx/SendTx/SignTx/SignTxForm.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ const SignTx = muiThemeable()((props) => {
       </div>
       <div style={{ paddingTop: '35px', display: 'flex', justifyContent: 'center' }}>
         <span style={{ color: props.muiTheme.palette.secondaryTextColor }}>
-          Plus a {trimEnd(fee.getDecimalized(), '0')} ETC fee for 21000 GAS
+          Plus a {trimEnd(fee.getDecimalized(), '0')} gas fee.
```
Comment (gagarin55) [inline src/components/tx/SendTx/SignTx/SignTxForm.js]: Fee is ETC not GAS

File: src/components/tx/SendTx/CreateTx/create.js [hunk 2]
```diff
@@ -92,56 +92,6 @@ const CreateTx = connect(
     onCancel: () => {
       ownProps.goDashboard();
     },
-    onSubmit: (data) => {
```
Comment (gagarin55) [inline src/components/tx/SendTx/CreateTx/create.js]: Tx creation should be located in CreateTx component

Comment (shanejonas) [reply src/components/tx/SendTx/CreateTx/create.js]: this wasn't actually modifying anything, all the tx create stuff happens in MultiCreateTx

File: src/containers/Screen/screen.js [hunk 3]
```diff
@@ -23,10 +22,14 @@ import PaperWallet from '../PaperWallet';
...
 import GenerateAccount from '../../components/accounts/GenerateAccount';
 
+import { Wei } from 'emerald-js';
 import WalletScreen from '../../store/wallet/screen';
+import MultiCreateTransaction from '../MultiCreateTransaction';
 
 const log = createLogger('screen');
 
+const mockOwnAddresses = ['0x00', '0x03', '0x004'];
+const mockAddressBookAddresses = ['0x00', '0x0111', '0x006'];
```
Comment (shanejonas) [inline src/containers/Screen/screen.js]: dead code

File: src/setupTests.js [hunk 4]
```diff
@@ -1 +1,7 @@
+const Enzyme = require('enzyme');
+const EnzymeAdapter = require('enzyme-adapter-react-16');
+
+// Setup enzyme's react adapter
+Enzyme.configure({ adapter: new EnzymeAdapter() });
```
Comment (shanejonas) [inline src/setupTests.js]: this is required to update enzyme to latest + support react 16

File: src/components/accounts/AccountShow/show.js [hunk 5]
```diff
@@ -95,30 +97,30 @@ export class AccountShow extends React.Component {
                 </div>
                 <div style={ styles.right }>
                   {!this.state.edit && <AddressAvatar
-                    editable
-                    addr={account.get('id')}
-                    description={account.get('description')}
-                    name={account.get('name')}
-                    onEditClick={this.handleEdit}
+                                         editable
```
Comment (BelfordZ) [inline src/components/accounts/AccountShow/show.js]: this spacing though

File: src/containers/MultiCreateTransaction/MultiCreateTransaction.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  render() {
+    return (
+      <Page title="Create Transaction" leftIcon={<Back onClick={this.props.goDashboard}/>}>
+        {this.getPage()}
+      </Page>
+    )
+  }
+}
+
+const ThemedCreateTransaction = muiThemeable()(MultiCreateTransaction);
```
Comment (shanejonas) [inline src/containers/MultiCreateTransaction/MultiCreateTransaction.js]: dont need this anymore

File: src/components/accounts/GenerateAccount/PasswordDialog/passwordDialog.js [hunk 7]
```diff
@@ -47,8 +49,8 @@ class PasswordDialog extends React.Component {
 
   onPassphraseChange = (newValue) => {
     const passphraseError = (newValue.length === 0 || newValue.length >= MIN_PASSWORD_LENGTH) ?
-      null :
-      this.state.passphraseError;
+                            null :
+                            this.state.passphraseError;
```
[the remark this answers is not in the document]
Comment (BelfordZ) [reply src/components/accounts/GenerateAccount/PasswordDialog/passwordDialog.js]: yep

File: src/components/tx/SendTx/SignTx/SignTx.test.js [hunk 8]
```diff
@@ -1,32 +1,30 @@
...
+     * tx = {
+     *   amount: convert.toBigNumber('100.0120'),
+     * };
 
-    component = shallow(<SignTx muiTheme={mockMuiTheme} tx={tx} fee={fee} />);
-    expect(component).toBeDefined();
-    expect(component.findWhere((n) => n.text() === '100.012')).toHaveLength(1);
+     * component = shallow(<SignTxForm muiTheme={mockMuiTheme} tx={tx} fee={fee} />);
+     * expect(component).toBeDefined();
+     * expect(component.findWhere((n) => n.text() === '100.012')).toHaveLength(1);*/
```
Comment (BelfordZ) [inline src/components/tx/SendTx/SignTx/SignTx.test.js]: should make an issue to fix this

File: src/components/tx/SendTx/SignTx/SignTxForm.js [hunk 9]
```diff
@@ -53,11 +51,15 @@ const displayFlexCenter = {
...
+const SignTx = muiThemeable()((props) => {
+  const { value, fiatRate, fiatCurrency, fee, tx, gas, token } = props;
+  const { onCancel, onChangePassword, onSubmit, useLedger } = props;
 
-  // const USDValue = Currency.format(Currency.convert(value, fiatRate, 2), fiatCurrency);
+  const onChange = (event, value) => {
+    onChangePassword(value);
+  }
+
+  //const USDValue = Currency.format(Currency.convert(tx.amount, fiatRate, 2), fiatCurrency);
```
Comment (BelfordZ) [inline src/components/tx/SendTx/SignTx/SignTxForm.js]: add this back in

File: src/containers/MultiCreateTransaction/MultiCreateTransaction.js [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+
+      const toAmount = etherToWei(parseFloat(transaction.amount).toFixed());
+      const gasLimit = new Wei(parseInt(transaction.gasLimit, 10)).getValue();
+      const gasPrice = new Wei(parseInt(transaction.gasPrice, 10)).getValue();
+
+      if (!tokenInfo) {
+        throw new SubmissionError(`Unknown token ${data.token}`);
+      }
+      // TODO: moved tx creation to CreateTransaction handler
+      // 1. Create TX here
```
Comment (BelfordZ) [inline src/containers/MultiCreateTransaction/MultiCreateTransaction.js]: remove comment?

**Pull request merged**